lib/fake-xhr/index.js

```diff
@@ -553,9 +553,11 @@ function fakeXMLHttpRequestFor(globalScope) {
                 this.dispatchEvent(
                     new sinonEvent.ProgressEvent(event, progress, this)
                 );
+                this.dispatchEvent(readyStateChangeEvent);
                 this.dispatchEvent(
                     new sinonEvent.ProgressEvent("loadend", progress, this)
                 );
+                return;
             }
 
             this.dispatchEvent(readyStateChangeEvent);
```

Comment:
I think I'd move the order of things around to avoid the surprising `return` at the end. What about something like

```javascript
if (this.readyState !== FakeXMLHttpRequest.DONE) {
    this.dispatchEvent(readyStateChangeEvent);
} else {
   // big block with no return at the end
}
```

Reply:
Alternatively you could extract the logic to make it clearer
```javascript
if (this.readyState === FakeXMLHttpRequest.DONE) {
    return this.handleXHRDone();
else {
    this.dispatchEvent(readyStateChangeEvent);
}

Reply:
No problem, I heed to your project codestyle and I appreciate the suggestions!  I prefer suggestion 1 so I'll push and update momentarily.

Reply:
request to resolve due to 183eed9.